tools/range_read/range_read.c

```diff
@@ -305,49 +326,40 @@ cursor(
                       be64toh(*(uint64_t *)key),
                       be64toh(*((uint64_t *)key + 1)));
         }
+        t_read = get_time_ns();
 
-        kh_cursor_destroy(c);
+        if (!opts.use_update)
+            kh_cursor_destroy(c);
 
-        if ((opcnt % dt->dt_skip == 0) && dt_idx < dt_size)
-            dt->dt[dt_idx++] = get_time_ns() - start;
+        t_full = get_time_ns();
+
+        hdr_record_value(lat->lat_create, t_create - t_start);
+        hdr_record_value(lat->lat_seek, t_seek - t_create);
+        hdr_record_value(lat->lat_read, (t_read - t_seek) / (i ?: 1));
+        hdr_record_value(lat->lat_full, t_full - t_start);
 
         if (++ncursor % 128 == 0)
             atomic_add(&n_cursor, 128);
     }
     atomic_add(&n_read, nread & 1023);
     atomic_add(&n_cursor, ncursor & 127);
 
-    if (dt)
-        dt->dt_cnt = dt_idx;
-
     free(vbuf);
+
+    if (opts.use_update) {
+        int i;
+
+        for (i = 0; i < NELEM(cur); i++)
+            kh_cursor_destroy(cur[i]);
+    }
 }
 
 void
-print_stats(
-    void *arg)
+print_stats()
 {
-    struct thread_arg    *targ = arg;
     uint32_t second = 0;
     uint64_t nw, nr, nc;
     uint64_t last_writes = 0, last_reads = 0, last_cursors = 0;
-    char logfile[2048];
-    FILE *logfd;
-
-    snprintf(logfile, sizeof(logfile), "%s/%s", opts.logdir, (char *)targ->arg);
-    logfd = fopen(logfile, "w");
-    if (!logfd)
-        fatal(errno, "Could not open log file %s", logfile);
-
-    fprintf(logfd, "\n%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s\n",
-            "timestamp", opts.vsep,
-            "elapsed", opts.vsep,
-            "writes", opts.vsep,
-            "reads", opts.vsep,
-            "cursors", opts.vsep,
-            "lRate", opts.vsep,
-            "rRate", opts.vsep,
-            "cRate");
```

Comment:
Incremental stats are more interesting IMO.  Normalized to per/second or total count since previous row.

Reply:
Yes, that's what the xRate stats are